platform/api/aws/ec2.go

```diff
@@ -255,12 +283,192 @@ func (a *API) getSecurityGroupID(name string) (string, error) {
 	return *sgIds.SecurityGroups[0].GroupId, nil
 }
 
+// getVPCID gets a VPC for the given security group
+func (a *API) getVPCID(sgId string) (string, error) {
+	sgs, err := a.ec2.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+		GroupIds: []*string{&sgId},
+	})
+	for _, sg := range sgs.SecurityGroups {
+		if sg.VpcId != nil {
+			return *sg.VpcId, nil
+		}
+	}
+	return "", fmt.Errorf("no vpc found for security group %v: %v", sgId, err)
+}
+
+// getSubnetID gets a subnet for the given VPC.
+func (a *API) getSubnetID(vpc string) (string, error) {
+	subIds, err := a.ec2.DescribeSubnets(&ec2.DescribeSubnetsInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("vpc-id"),
+				Values: []*string{&vpc},
+			},
+		},
+	})
+	if err != nil {
+		return "", fmt.Errorf("unable to get subnets for vpc %v: %v", vpc, err)
+	}
+	for _, id := range subIds.Subnets {
```

Comment:
nit: Since this will get the first subnet id should the method be named ``getFirstSubnetID``?

Reply:
This functions the same as the pre-existing `getSecurityGroupID` which also will return the first entry if one exists (otherwise one is created)

Reply:
OK, fair enough.